base/errorshow.jl

```diff
@@ -160,17 +160,6 @@ showerror(io::IO, ex::UndefKeywordError) =
 
 function showerror(io::IO, ex::UndefVarError)
     print(io, "UndefVarError: $(ex.var) not defined")
-    Experimental.show_error_hints(io, ex)
```

Comment:
Can we keep that line? The goal of #39850 was to allow packages to add hints similar to the LegacyStrings one.

Reply:
I suppose that doesn't hurt? I don't know that the API premise of `Experimental.show_error_hints` is, but I suppose an InteractiveUtils or REPL package might decide to spell check these or something

Reply:
As far as API premises go, I think we probably just want to document the kinds of errors `register_error_hint!` supports and add them as users request them. We don't have that many error types anyways.